enable compiling using the Kbuild subsystem

example:

make ARCH=arm CROSS_COMPILE=armv5b-softfloat-linux- -C /usr/src/linux-nslu2 M=`pwd`

this is a kbuild only patch - use Makefile.patch on OpenEmbedded

 Kbuild |   33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

--- ixp400-eth/Kbuild	1970-01-01 00:00:00.000000000 +0000
+++ ixp400-eth/Kbuild	1970-01-01 00:00:00.000000000 +0000
@@ -0,0 +1,33 @@
+obj-m := ixp400_eth.o
+
+OSAL_DIR	:= ../ixp_osal
+IX_XSCALE_SW	:= ../ixp400_xscale_sw
+
+CFLAGS := -D __KERNEL__ -D__OPTIMIZE__ -Wall -Wno-trigraphs -fno-common -pipe \
+		-mapcs-32 -msoft-float -D__linux -DCPU=33 -DXSCALE=33 \
+		-D__LINUX_ARM_ARCH__=5 -mtune=xscale
+
+# includes
+CFLAGS += -Iinclude\
+	-I$(IX_XSCALE_SW)/src/include\
+	-I$(OSAL_DIR)/\
+	-I$(OSAL_DIR)/os/linux/include/\
+	-I$(OSAL_DIR)/os/linux/include/modules/\
+	-I$(OSAL_DIR)/os/linux/include/modules/ioMem/\
+	-I$(OSAL_DIR)/os/linux/include/modules/core/\
+	-I$(OSAL_DIR)/os/linux/include/modules/bufferMgt/\
+	-I$(OSAL_DIR)/os/linux/include/core/\
+	-I$(OSAL_DIR)/os/linux/include/platforms/\
+	-I$(OSAL_DIR)/os/linux/include/platforms/ixp400/\
+	-I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ixp425\
+	-I$(OSAL_DIR)/os/linux/include/platforms/ixp400/ixp465\
+	-I$(OSAL_DIR)/os/linux/include/core/\
+	-I$(OSAL_DIR)/include/\
+	-I$(OSAL_DIR)/include/modules/\
+	-I$(OSAL_DIR)/include/modules/bufferMgt/\
+	-I$(OSAL_DIR)/include/modules/ioMem/\
+	-I$(OSAL_DIR)/include/modules/core/\
+	-I$(OSAL_DIR)/include/platforms/\
+	-I$(OSAL_DIR)/include/platforms/ixp400/\
+	-I$(OSAL_DIR)/include/platforms/ixp400/ixp425\
+	-I$(OSAL_DIR)/include/platforms/ixp400/ixp465
